### PR TITLE
make RandomPeer return an error

### LINKED ISSUE #140: RandomPeer should not panic
Really, there should always be a peer to select, even if it's yourself. But this is probably best practice anyway.

### PR DESCRIPTION
**Pull request by lukechampine**
This actually fixes #140.

### CODE REVIEW

File: network/network_test.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func TestPeerSharing(t *testing.T) {
 	peer := Address("foo:9001")
 	tcps.AddPeer(peer)
 	// tcps only has one peer, so RandomPeer() should return peer
-	if tcps.RandomPeer() != peer {
+	if p, err := tcps.RandomPeer(); err != nil || p != peer {
```
Comment (DavidVorick) [inline network/network_test.go]: IF thing1, thing2 := function1; statement1 || statement 2 {

pretty busy line

File: siad/network.go [hunk 2]
```diff
@@ -65,7 +65,10 @@ func (d *daemon) initializeNetwork(addr string, nobootstrap bool) (err error) {
 		// Every 2 minutes, call CatchUp() on a random peer. This helps with
 		// synchronization.
 		for {
-			go d.CatchUp(d.network.RandomPeer())
+			peer, err := d.network.RandomPeer()
+			if err == nil {
```
Comment (DavidVorick) [inline siad/network.go]: need to add a comment if you're leaving it `err == nil`

something like 
```go
if err == nil { // only call CatchUp if there was no error
}
```

would be fine

Comment (DavidVorick) [inline siad/network.go]: just something to acknowledge that convention is being broken

Comment (lukechampine) [inline siad/network.go]: how about this?
```go
for {
    time.Sleep(time.Minute * 2)
    peer, err := d.network.RandomPeer()
    if err != nil {
        continue
    }
    go d.CatchUp(peer)
}
```

Comment (DavidVorick) [inline siad/network.go]: it's weird that you can't just call `for time.Sleep(time.Minute * 2) {`, but yeah this looks fine to me.

**Pull request merged**